### PR TITLE
refactor: make SfMegaMenu more customizable

### PR DESCRIPTION
**Pull request by pspaczek**
# Scope of work

Support old API for this component but I recommended change implementation to the new common example base on SfMegaMenuColumns.
Now SfMegaMenu should be more customizable

# Screenshots of visual changes

Common
Zrzut ekranu 2020-05-21 o 15 44 57
Zrzut ekranu 2020-05-21 o 15 45 14
Zrzut ekranu 2020-05-21 o 15 45 21

Search Results
Zrzut ekranu 2020-05-21 o 15 48 11
Zrzut ekranu 2020-05-21 o 15 48 20

 

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))
 - [ ] I accept [Individual Contributor License Agreement](https://github.com/DivanteLtd/next/blob/master/CLA.md)

### CODE REVIEW

**Comment by martaradziszewska** [comment]
@pspaczek please resolve conflict :)

**Comment by pspaczek** [comment]
@martaradziszewska done :)

File: packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js [hunk 1]
```diff
@@ -2,343 +2,373 @@ import { storiesOf } from "@storybook/vue";
...
+      <SfMegaMenuColumn title="Categories" class="sf-mega-menu-column--pined-content-on-mobile" style="flex: 0 0 220px; --menu-item-mobile-nav-icon-display: none;">
+        <SfList>
+          <SfListItem v-for="(category, key) in categories">
+            {{category}}
+          </SfListItem>
+        </SfList>
+      </SfMegaMenuColumn>
+      <SfMegaMenuColumn title="Product suggestions" class="sf-mega-menu-column--pined-content-on-mobile" style="flex: 1">
+        <template #title="{title}">
+          <div style="display: flex; align-items: center; --menu-item-mobile-nav-icon-display: none;">
```
Comment (martaradziszewska) [inline packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js:144]: I think, bottom border should be on 100% width
image

File: packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js [hunk 2]
```diff
@@ -2,343 +2,373 @@ import { storiesOf } from "@storybook/vue";
...
+            <SfMenuItem
+                :label="title"
+                class="sf-mega-menu-column__header"
+            />
+            <span style="margin: 0 0 0 var(--spacer-sm); color:var(--c-gray); font: var(--font-normal) var(--font-sm)/1.4 var(--font-family-secondary);">(20 items)</span>
+          </div>
+        </template>
+        <SfScrollable style="--scrollable-max-height: 630px;" show-text="" hide-text="">
+          <div style="display: flex; flex-wrap: wrap;">
```
Comment (martaradziszewska) [inline packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js:153]: Can we center search results? 
image
or make 2 columns as in designs:
image

File: packages/vue/config/storybook/stories.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
+@mixin for-mobile {
```
[the remark this answers is not in the document]
Comment (pspaczek) [reply packages/vue/config/storybook/stories.scss:1]: IMO we can do it and use css instead inline styles

File: packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js (hunk 2 above)
Comment (pspaczek) [reply packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js:153]: Yes, but IMO we can do it on another issue :)

File: packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js (hunk 1 above)
Comment (pspaczek) [reply packages/vue/src/components/organisms/SfMegaMenu/SfMegaMenu.stories.js:144]: Zrzut ekranu 2020-05-22 o 16 00 40
fixed, thank You 💚

**Pull request merged**